### ISSUE TITLE
MUI Premium License extension and component compatibility across versions

### ISSUE THREAD

**Issue by soodaws** [mui/mui-x#10320]
### Order ID

51042

### The problem in depth 🔍

Hey MUI team. We have a peculiar situation around licensing and component compatibility that I’m looking for your input on.   

### Context 
Our team has a premium license which was bought for a UI package A. This UI package is on the following versions of the mui libraries.  
  
  **Package A's, package.json** 

    "@mui/material": "5.4.4",  
    "@mui/styled-engine-sc": "5.10.1",  
    "@mui/x-license-pro": "6.0.4",  
    "@mui/x-data-grid-premium": "6.2.1",

We have a need to use `@mui/x-data-grid-pro` in package B. The problem however is that package B is on mui v4. 

  **Package B's, package.json** 

    "@material-ui/core": "^4.11.3",
    "@material-ui/data-grid": "^4.0.0-alpha.30",

### Questions

1. MUI's [documentation](https://mui.com/x/introduction/licensing/#license-key-installation) says that the newer license keys are only compatible with mui x from v5 onwards. Will be able to use the existing license key on package B given that package B is on MUI core v4? If not, and we need to get a separate license key for v4, will we need to re-pay and get another subscription for this new key or can we get that key as an extension of our existing key?

2. In package B, the need is to use `@mui/x-data-grid-pro`. If we want to use the `v6.x.x` version of the data grid, it seems like we have to do some additional to make `6.x.x components` work with `material-ui/core v4`. 
The [documentation](https://mui.com/x/migration/migration-data-grid-v4/) goes over some steps for making MUI Core v4 work with MUI X v5, but doesn't talk about the feasibility of making MUI Core v4 work with MUI X v6. I'm assuming here that MUI X versions are synonymous to the `data-grid-pro` version we will use. So this question is about feasibility - will we be able to use `@mui/x-data-grid-pro v6` with MUI Core v4? There are ofcourse API changes to the DataGrid that we will need to take care of, but other than that is there any other constraint or additional steps we need to take? 

**To summarize our options based on the information I've gathered are -** 

1. Get a license key for v4(with or without paying again)
2. Use data-grid pro v4 in package B without any additional compatibility changes to packageB or
3. Use data-grid pro v6 in package B, with compatibility changes between mui core v4 and mui x v6 (if feasible) and api changes. 

Would appreciate if you can shed some light on the questions and on what the most efficient way forward would be in your opinion. 

Thank you. 

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by DanailH** [comment]
@soodaws it will be difficult to answer that one here. I would suggest you open a support request via this form https://support.mui.com/hc/en-us/requests/new?tf_360023797420=mui_x and explain again the issue. GitHub is met for technical problems and this is a more business-related issue.

That being said I'm tagging @joserodolfofreitas.Maybe he will be able to hep.

**Comment by romgrk** [comment]
Not sure Danail's tag worked: cc @joserodolfofreitas
